tchannel/tornado/dispatch.py

```diff
@@ -182,9 +185,11 @@ def handle_call(self, request, connection):
                 )
 
                 # get new top-level resp from controller
-                new_resp = yield gen.maybe_future(
-                    handler.endpoint(new_req)
-                )
+                with StackContext(lambda: _context):
+                    tchannel.get_context().parent_tracing = request.tracing
+                    f = handler.endpoint(new_req)
+
```

Comment:
This should all be encapsulated in that same module that manages the thread. Ideally, this code block should just be

    with Context(parent_tracing):
        # ...

(With a better name than Context, of course.)